src/main/scala/strawman/collection/View.scala

```diff
@@ -13,6 +15,20 @@ trait View[+A] extends Iterable[A] with IterableOps[A, View, View[A]] {
     fromIterable(coll)
 
   override def className = "View"
+
+  def groupBy[K](f: (A) => K): immutable.Map[K, View[A]] = {
+    val m = mutable.Map.empty[K, ArrayBuffer[A]]
+    for (elem <- coll) {
+      val key = f(elem)
+      val bldr = m.getOrElseUpdate(key, ArrayBuffer.empty)
```

Comment:
I think a normal `ArrayBuffer` allocates too much memory.  If there isn't much redundancy you could end up increasing the memory usage by about 5x (!).  I am not sure we have a great alternative data structure right now, but we should think about one.  At the very least I think we should have an array builder and a view of the built array so at *least* we can reclaim all the extra memory after construction is complete.